client/my-sites/marketing/ultimate-traffic-guide/index.js

```diff
@@ -0,0 +1,289 @@
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { Button, CompactCard } from '@automattic/components';
+import {
+	getProductDisplayCost,
+	isProductsListFetching,
+} from 'calypso/state/products-list/selectors';
+import QueryUserPurchases from 'calypso/components/data/query-user-purchases';
+import { isFetchingUserPurchases, getUserPurchases } from 'calypso/state/purchases/selectors';
+import { getCurrentUserId } from 'calypso/state/current-user/selectors';
+import { getSiteSlug } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
+import {
+	DefaultVariation,
+	Experiment,
+	LoadingVariations,
+	Variation,
+} from 'calypso/components/experiment';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+export const WPCOM_TRAFFIC_GUIDE = 'traffic-guide';
+
+export const hasTrafficGuidePurchase = ( purchases ) =>
+	!! (
+		purchases && purchases.find( ( purchase ) => WPCOM_TRAFFIC_GUIDE === purchase.productSlug )
+	);
+
+export const downloadTrafficGuide = () => {
+	window.location.href = 'https://public-api.wordpress.com/wpcom/v2/traffic-guide-download';
+};
```

Comment:
Not sure if this is the best place for these exported consts

Reply:
The main problem is with the WPCOM_TRAFFIC_GUIDE constant -  it looks like Calypso doesn't have a centralized place or a unified standard for naming such constants. For example, the plans slugs are in lib/plans/constants.js and transfers are in lib/domains/constants. Jetpack products are in [lib/products-values/constants.js](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/products-values/constants.js)

Should you create lib/ebooks/constants for this specific constant? Not sure. Maybe. An alternative would be to simply hardcode the product slug instead of use a constant, [like the Concierge Session does](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/products-values/is-concierge-session.js).

If it was me, I'd probably use the constant and put it where the Jetpack constants are. Looks like Jetpack folks tried to create a unified place but stopped after adding their constants and nobody else picked it up.